pkg/serviceaccount/legacy.go

```diff
@@ -126,6 +140,32 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 			klog.V(4).Infof("Service account UID no longer matches %s/%s: %q != %q", namespace, serviceAccountName, string(serviceAccount.UID), serviceAccountUID)
 			return nil, fmt.Errorf("ServiceAccount UID (%s) does not match claim (%s)", serviceAccount.UID, serviceAccountUID)
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyServiceAccountTokenTracking) {
+			warning.AddWarning(ctx, "", "Use tokens from the TokenRequest API or manually created secret-based tokens instead of auto-generated secret-based tokens.")
+			today := time.Now().Format("2006-01-02")
+			if secret.GetAnnotations()[LastUsedAnnotationKey] != today || secret.GetLabels()[InUseLabelKey] != today {
+				if secret.GetAnnotations() == nil {
+					secret.SetAnnotations(map[string]string{})
+				}
+				secret.GetAnnotations()[LastUsedAnnotationKey] = today
+				if secret.GetLabels() == nil {
+					secret.SetLabels(map[string]string{})
+				}
+				secret.GetLabels()[InUseLabelKey] = today
```

Comment:
drop these lines, we're using the patch to modify the secret instead of modifying it with an update

Reply:
done